### ISSUE TITLE
Batch norm forward accuracy issue (7% gap with PT2.6)

### ISSUE THREAD

**Issue by xytintel** [intel/torch-xpu-ops#1429]
Reproducer:

```python
import torch
import torch.nn as nn

N, C, H, W = 256, 256, 56, 56

std = 100
torch.manual_seed(1234)
x = torch.randn(N, C, H, W).to(memory_format=torch.channels_last) * std
x_xpu = x.clone().detach_().xpu().to(memory_format=torch.channels_last) * std

x.requires_grad = True
x_xpu.requires_grad = True

bn = nn.BatchNorm2d(C)
bn_xpu = nn.BatchNorm2d(C).xpu()
bn_xpu.weight.data[:] = bn.weight
bn_xpu.bias.data[:] = bn.bias

out = bn(x).mean()
out.backward()

out_xpu = bn_xpu(x_xpu).mean()
out_xpu.backward()

diff1 = (out - out_xpu.cpu()).abs()
diff2 = (x.grad - x_xpu.grad.cpu()).abs()
max_diff1 = diff1.max().item()
max_diff2 = diff2.max().item()
median_diff1 = diff1.median().item()
median_diff2 = diff2.median().item()

print('max_diff1:', max_diff1, 'median:', median_diff1)
print('max_diff2:', max_diff2, 'median:', median_diff2)

# prof_xpu = torch.profiler.profile(
#     activities=[
#     torch.profiler.ProfilerActivity.CPU,
#     torch.profiler.ProfilerActivity.XPU],
# )
# with prof_xpu:
#     for i in range(128):
#         output = bn_xpu(x_xpu)
#         output.backward(x_xpu)
# print(prof_xpu.key_averages(group_by_input_shape=True).table(sort_by="self_xpu_time_total", row_limit=100000))
# print(output.dtype)

# ref-2.6
# max_diff1: 4.1814482432123157e-10 median: 4.1814482432123157e-10
# max_diff2: 1.0513642602200068e-17 median: 5.452609161827048e-18
# target-2.7
# max_diff1: 4.466546854597908e-10 median: 4.466546854597908e-10
# max_diff2: 1.0513642602200068e-17 median: 5.4526070938755164e-18
```

### Versions

2.7

**Comment by xytintel** [comment]
Not a issue
